nifi-extension-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/AbstractMongoProcessor.java

```diff
@@ -235,4 +268,20 @@ protected synchronized void configureMapper(String setting, String dateFormat) {
             objectMapper.setDateFormat(df);
         }
     }
+
+    /**
+     * Checks if given update mode option matches for the incoming flow file
+     * @param updateModeToMatch the value against which processor's mode is compared
+     * @param processorMode the value coming from running processor
+     * @param flowFile incoming flow file to extract processor mode
+     * @return true if the incoming files update mode matches with updateModeToMatch
+     */
+    public boolean updateModeMatches(
```

Comment:
This should be `protected`:
```suggestion
    protected boolean updateModeMatches(
```